src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AuthenticatorFactoryInterface.php

```diff
@@ -24,4 +27,14 @@ interface AuthenticatorFactoryInterface
      * @return string|string[] The authenticator service ID(s) to be used by the firewall
      */
     public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId);
+
+    /**
+     * Defines the configuration key used to reference the authenticator
+     * in the firewall configuration.
+     *
+     * @return string
```

Comment:
Can we use proper return type instead?

Reply:
Unfortunately not. These 2 methods are a copy past of `SecurityFactoryInterface` and this interface is already in use by all factories providing support for the new system. If we add this typehint now, it will be a BC break for them.

We'll have to do the same as all other return types here: introduce it in the 6.0 branch.